cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/provisioner/TetheringConf.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.tethering.runtime.spi.provisioner;
+
+import java.util.Map;
+
+/**
+ * Configuration for the Tethering Provisioner.
+ */
+public class TetheringConf {
+  private final String tetheredInstanceName;
+  private final String tetheredNamespace;
+
+  private TetheringConf(String tetheredInstanceName, String tetheredNamespace) {
+    this.tetheredInstanceName = tetheredInstanceName;
+    this.tetheredNamespace = tetheredNamespace;
+  }
+
+  /**
+   * Create the conf from a property map while also performing validation.
```

Comment:
should we check if the tethered instance/namespace exist by querying the tethering store?

Reply:
I didn't originally think about that since the UI will be using an API to pull the tethered namespace/instance info, but that sounds like a good idea to validate that the connection still exists / state hasn't changed